test/unit/storage_test.rb

```diff
@@ -138,12 +185,15 @@ def assert_sentinel_connector(client)
     assert_instance_of Redis::Client::Connector::Sentinel, connector
   end
 
-  def assert_client_config(client, host: nil, port: nil, url: nil, sentinels: nil)
+  def assert_client_config(client, host: nil, port: nil, url: nil, **opts)
     raise "bad usage of #{__method__}" unless host || port || url
     assert_equal client.port, port if port
     assert_equal client.host, host if host
     assert_equal client.options[:url], url if url
+    sentinels = opts[:sentinels]
     assert_equal client.options[:sentinels], sentinels if sentinels
+    role = opts[:role]
+    assert_equal client.options[:role], role if role
```

Comment:
how about:
```ruby
opts.each do |o, v|
  assert_equal client.options[o], v
end
```
This would also allow you to pass `url` as part of `**opts`: `def assert_client_config(client, host: nil, port: nil, **opts)`

Reply:
Host and port are no longer used in this test set. So, refactoring makes sense.

Reply:
done